### PR TITLE
improve heuristic for when to refresh line in the REPL

### PR DESCRIPTION
**Pull request by KristofferC**
Comparing pasting the same long function in the REPL before PR (above) and after PR (below): https://media.giphy.com/media/3o6Ztmlq9XLVMTqDgA/giphy.gif. The before PR version was going quite a while after I turned of the recording. Still not perfect, but better.

cc @vtjnash @Keno 

Sort of fixes [no auto close] https://github.com/JuliaLang/julia/issues/15787

### CODE REVIEW

File: base/LineEdit.jl [hunk 1]
```diff
@@ -447,10 +447,18 @@ function edit_replace(s, from, to, str)
...
+        seek(buf, p)
+        return line_size
+    end
     str = string(c)
-    edit_insert(s.input_buffer, str)
-    if !('\n' in str) && eof(s.input_buffer) &&
-        ((position(s.input_buffer) + sizeof(s.p.prompt) + sizeof(str) - 1) < width(terminal(s)))
+    edit_insert(buf, str)
+    if !('\n' in str) && eof(buf) && 
+        ((line_size(s) + sizeof(s.p.prompt) + sizeof(str) - 1) < width(terminal(s)))
```
Comment (vtjnash) [inline base/LineEdit.jl]: `sizeof(s.p.prompt)` only applies to the first line. the other lines use `s.indent`

Comment (KristofferC) [inline base/LineEdit.jl]: In case someone somewhere has  `s.indent != strwidth(s.p.prompt)`?

Comment (KristofferC) [inline base/LineEdit.jl]: Anyway, updated.

**Comment by KristofferC** [comment]
Any further comments here? @vtjnash for the last commit with `s.indent`?

**Pull request merged**